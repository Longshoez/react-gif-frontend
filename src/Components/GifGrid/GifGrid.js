import React from 'react';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import GifContainer from '../GifContainer/GifContainer';
import 'animate.css'
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

    const { data:images, loading } = useFetchGifs( category )

  return (
      <div>
          <h2 className='searchText'>{category}</h2>          
          {loading && <p className='searchText animate__animated animate__flash'>Loading</p>}
          <ol className='gifSearchContainer'>
              {images.map(gif => (
                  <GifContainer title={gif.title} url={gif.url} key={gif.id}/>
              ))}
          </ol>
      </div>
  );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}