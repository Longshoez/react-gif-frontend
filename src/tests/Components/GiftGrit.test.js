import React from 'react';
import {shallow} from 'enzyme';
import { GifGrid } from '../../Components/GifGrid/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs') //mocks file calls and controlls data sent and recieved

describe('Testing GifGrid Component', () => {

    const category = "Something"
    useFetchGifs.mockReturnValue({
        data: [],
        loading: true,
    })
    let wrapper = shallow(<GifGrid category={category}/>)


    test('should render Component', () => {                        
        expect(wrapper).toMatchSnapshot()
    });

    test('should show items when useFetchGids is loaded', () => {
    //Arrange
    const gifs = [{
        id: 'ABC',
        url: 'https://localhost/yeah/yeah.jpg',
        title: 'some some'
    }]
    
    //Act
    useFetchGifs.mockReturnValue({
        data: gifs,
        loading: false,
    })
    let wrapper = shallow(<GifGrid category={category}/>)
    
    //Assert    
    //expect(wrapper).toMatchSnapshot()
    //no p element should exist if gifs are already fetched    
    expect(wrapper.find('p').exists()).toBe(false) 
    expect(wrapper.find('GifContainer').length).toBe(gifs.length)

    });
    
});
