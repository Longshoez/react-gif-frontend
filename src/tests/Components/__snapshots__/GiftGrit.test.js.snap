// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid Component should render Component 1`] = `
<div>
  <h2
    className="searchText"
  >
    Something
  </h2>
  <p
    className="searchText animate__animated animate__flash"
  >
    Loading
  </p>
  <ol
    className="gifSearchContainer"
  />
</div>
`;
